server/models/groups/getGroup.js

```diff
@@ -0,0 +1,48 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {Object} param0.params
+ * @return {Promise}
+ *
+ */
+
+function getGroup({ username, params }) {
+  return new Promise((resolve, reject) => {
+    const { group_id: groupId } = params
+    pool.query(
+      `SELECT user_groups.is_group_moderator, \`groups\`.confidential FROM user_groups 
+      INNER JOIN \`groups\` ON  user_groups.group_id = \`groups\`.id
+      WHERE user_groups.group_id=? AND user_groups.username=?`,
+      [groupId, username],
+      (error, results) => {
+        if (error) {
+          return reject(error)
+        }
+        const { is_group_moderator, confidential } = results[0]
+        if (is_group_moderator === 0 && confidential === 1) {
+          return reject(
+            'The group is confidential and user do not have moderator access!'
+          )
+        }
+        pool.query(
+          `SELECT users.username, users.full_name, users.admission_number, users.profile_img, users.email, users.course, users.department, 
+          user_groups.is_group_moderator FROM user_groups 
+          INNER JOIN users ON user_groups.username = users.username
+          WHERE user_groups.group_id=? ORDER BY users.admission_number`,
+          [groupId],
+          (error, groupResults) => {
+            if (error) {
+              return reject(error)
+            }
+            return resolve(groupResults)
+          }
+        )
+      }
```

Comment:
Here, we are performing 2 joins, which seems to have a very high time complexity. Is  there any other better way to do it ?

Reply:
I think the first JOIN will have constant complexity

Reply:
because.... ```INNER JOIN \`groups\` ON  user_groups.group_id = \`groups\`.id``` in this, user_groups.group_id = groups.id will match a unique entry

Reply:
Although, the time complexity would be greater, seems that there is no other way for this. So, we can keep this.